slither/utils/upgradeability.py

```diff
@@ -273,7 +409,7 @@ def encode_ir_for_compare(ir: Operation) -> str:
     if isinstance(ir, Assignment):
         return f"({encode_var_for_compare(ir.lvalue)}):=({encode_var_for_compare(ir.rvalue)})"
     if isinstance(ir, Index):
-        return f"index({ntype(ir.index_type)})"
+        return f"index({ntype(ir.variable_right.type)})"
```

Comment:
It only became wrong when the `index_type` property was removed from the `Index` class. Slither was just crashing when I tried to access the property, since it didn't exist anymore. Is that still something we can test?